apis/v1alpha1/opampbridge_types.go

```diff
@@ -0,0 +1,137 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// OpAMPBridgeSpec defines the desired state of OpAMPBridge.
+type OpAMPBridgeSpec struct {
+	// OpAMP backend Server endpoint
+	// +required
+	Endpoint string `json:"endpoint"`
+	// Protocol scheme for the communication between OpAMP server and OpAMP Bridge
+	// +required
+	Protocol string `json:"protocol"`
+	// Capabilities supported by the OpAMP Bridge
+	// +required
+	Capabilities []string `json:"capabilities"`
+	// ComponentsAllowed is a list of allowed OpenTelemetry components for each pipeline type (receiver, processor, etc.)
+	// +optional
+	ComponentsAllowed map[string][]string `json:"components_allowed,omitempty"`
+	// Resources to set on the OpAMPBridge pods.
+	// +optional
+	Resources v1.ResourceRequirements `json:"resources,omitempty"`
+	// NodeSelector to schedule OpAMPBridge pods.
+	// +optional
+	NodeSelector map[string]string `json:"nodeSelector,omitempty"`
+	// Replicas is the number of pod instances for the OpAMPBridge.
+	// +optional
+	Replicas *int32 `json:"replicas,omitempty"`
```

Comment:
I think we have to initially enforce that only one replica can be run at a time. Eventually we should figure out a high availability strategy for this (probably involving leader election)

Reply:
ah cool! Maximum=1, nifty :)